File: Python/0645.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        n = len(nums)
        count = [0] * (n + 1)

        missing, duplicate = 0, 0

        for num in nums:
            count[num] += 1

        for i in range(1, len(count)):
            if count[i] == 2:
                duplicate = i
            if count[i] == 0:
                missing = i

        return [duplicate, missing]

File: C/0645.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */

#include <stdio.h>

int* findErrorNums(int* nums, int numsSize, int* returnSize) {
    int count[numsSize + 1] = {};
    int duplicate = 0, missing = 0;

    for (int i = 0; i < numsSize; i++) {
        count[nums[i]] += 1;
    }

    for (int i = 1; i < numsSize + 1; i++) {
        if (count[i] == 2) duplicate = i;
        if (count[i] == 0) missing = i;
    }

    int *ans = malloc(2 * sizeof(int));
    ans[0] = duplicate;
    ans[1] = missing;
    *returnSize = 2;

    return ans;
}
